ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -0,0 +1,762 @@
+        result = run_migrate(
+            self.replicas[0],
+            "stage-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n'],
+        )
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert result.returncode == 0
+        assert MIGRATION_SCHEMA_LOG_MSG in install_msg
+        assert MIGRATION_CONFIG_LOG_MSG in install_msg
+        assert IPA_UPGRADE_LOG_MSG in install_msg
+        assert SIDGEN_TASK_LOG_MSG in install_msg
+        assert MIGRATION_COMPLETE_LOG_MSG in install_msg
+
+    def test_ipa_migrate_prod_mode(self, empty_log_file):
+        """
+        This test checks that ipa-migrate is successful
+        in prod run mode
+        """
+        tasks.kinit_admin(self.master)
+        tasks.kinit_admin(self.replicas[0])
+        MIGRATION_SCHEMA_LOG_MSG = "Migrating schema ...\n"
+        MIGRATION_DATABASE_LOG_MSG = (
+            "Migrating database ... (this make take a while)\n"
+        )
+        IPA_UPGRADE_LOG_MSG = (
+            "Running ipa-server-upgrade ... (this make take a while)\n"
+        )
+        SIDGEN_TASK_LOG_MSG = "Running SIDGEN task ...\n"
+        result = run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n'],
+        )
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert result.returncode == 0
+        assert MIGRATION_SCHEMA_LOG_MSG in install_msg
+        assert MIGRATION_DATABASE_LOG_MSG in install_msg
+        assert IPA_UPGRADE_LOG_MSG in install_msg
+        assert SIDGEN_TASK_LOG_MSG in install_msg
+
+    def test_ipa_migrate_with_bind_pwd_file_option(self, empty_log_file):
+        """
+        This testcase checks that ipa-migrate tool
+        works with valid bind_pwd specified in a file using '-j'
+        option
+        """
+        DEBUG_MSG = "--bind-pw-file=/tmp/pwd.txt\n"
+        bind_pwd_file = "/tmp/pwd.txt"
+        bind_pwd_file_content = self.master.config.admin_password
+        self.replicas[0].put_file_contents(
+            bind_pwd_file, bind_pwd_file_content
+        )
+        param = ['-j', bind_pwd_file, '-x']
+        result = run_migrate(
+            host=self.replicas[0],
+            mode="stage-mode",
+            remote_host=self.master.hostname,
+            bind_dn="cn=Directory Manager",
+            bind_pwd=None,
+            extra_args=param,
+        )
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert DEBUG_MSG in install_msg
+        assert result.returncode == 0
+
+   def test_ipa_migrate_using_db_ldif(self):
```

Comment:
ipatests/test_integration/test_ipa_ipa_migration.py:694: [E0001(syntax-error), ] Parsing failed: 'unindent does not match any outer indentation level (<unknown>, line 694)')